qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -93,20 +123,23 @@ def _compute_gradients(
             List of pairs consisting of gradient and excitation operator.
         """
         res = []
-        # compute gradients for all excitation in operator pool
         for exc in self._excitation_pool:
             # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
+            # the ansatz needs to be decomposed for the gradient to work
+            vqe.ansatz = self._ansatz.decompose()
+            param_sets = sorted(vqe.ansatz.parameters,key=lambda p: p.name)
+            #zip will only iterate the length of the shorter list
+            theta1=dict(zip(theta,vqe.ansatz.parameters))
```

Comment:
```suggestion
            theta1=dict(zip(vqe.ansatz.parameters, theta))
```
The arguments here need to be flipped because you want to map from parameter names to their optimal values (stored in theta).
This resolves the error you were showing to me on Slack